src/backend/distributed/executor/subplan_execution.c

```diff
@@ -56,18 +66,49 @@ ExecuteSubPlans(DistributedPlan *distributedPlan)
 	 */
 	BeginOrContinueCoordinatedTransaction();
 
-	nodeList = ActiveReadableNodeList();
-
 	foreach(subPlanCell, subPlanList)
 	{
 		DistributedSubPlan *subPlan = (DistributedSubPlan *) lfirst(subPlanCell);
 		PlannedStmt *plannedStmt = subPlan->plan;
 		uint32 subPlanId = subPlan->subPlanId;
+
 		DestReceiver *copyDest = NULL;
 		ParamListInfo params = NULL;
 		EState *estate = NULL;
 
 		char *resultId = GenerateResultId(planId, subPlanId);
+		IntermediateResultsHashEntry *entry = SearchIntermediateResult(
```

Comment:
I think my comment is lost: Can we move  this logic to a function where we give `resultId ` and get `nodeIdList`? If `nodeIdList == NIL`, we write to a local  file

Reply:
I misunderstood your previous comment. I created `SearchIntermediateResult` to address that comment. As I see it know there is some more room for improvement here. 

What you are suggesting here seems like an elegant and cleaner way to implement this. I will move the logic here to a seperate function

Reply:
Ended up having
`if (workerNodeList == NIL) { .. }`
as we have a list of `WorkerNode`s at this point.